### ISSUE TITLE
During a command with lots of output, holding the scrollbar will not scroll correctly

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#430]
Now that we can actually get this much output, it appears the scrollbar is less than ideal in this case. Holding the scrollbar in the middle of the buffer during a long output command will show the scrollbar jumping from the end of the buffer to the mouse position. When held, the scrollbar should take precedence if possible.

### Steps to reproduce

1. ls -lR ~
2. Hold scrollbar in the middle

**Comment by Tyriar** [comment]
Not sure we can fix this without having some scrollend event or some reliable way of knowing when the scroll bar is currently being held. The current behavior is ideal when you scroll via mouse wheel.

**Comment by parisk** [comment]
I think a good way of handling this would be:

1. listening to `mousedown` and `mouseup` events on the scrollbar
2. emit these events as `scrollbar-hold` and `scrollbar-release`
3. set and unset the "state" of the scrollbar in an attribute

I can take this over if it's 🆗  with you.

**Comment by Tyriar** [comment]
You can't listen to events on the scroll bar as far as I'm aware.

**Comment by parisk** [comment]
Yep that's true. I did a quick search though and found this article from @darkwing: https://davidwalsh.name/detect-scrollbar-width.

Seems like we can calculate the width of the scrollbar, and since we know it's position and the mouse's position we could calculate when it is being clicked.

The issue I ran into though is that I was not able to detect the click itself somehow. There are chances that click events on the scrollbar do not propagate back to the viewport or the `.xterm` container.
